test/functional/test_framework/test_framework.py

```diff
@@ -714,7 +714,7 @@ def _initialize_chain_clean(self):
     def skip_if_no_py3_zmq(self):
         """Attempt to import the zmq package and skip the test if the import fails."""
         try:
-            import zmq  # noqa
+            import zmq  # type:ignore
```

Comment:
This line appears to be getting caught in the linter. 

`test/functional/test_framework/test_framework.py:717:13: F401 'zmq' imported but unused`

Reply:
As far as I can tell, we don't be able to fix this short of disabling flake8 linting for the entire file.  It's a false positive, in this case the very act of importing the module is "using" it.

Reply:
It's worth noting that this brings us from 3 `F401` warnings up to 4 `F401` warnings.

Reply:
is it not possible to keep the `# noqa` linter disable comment?

Reply:
Unfortunately not.  We need the `type:ignore` or else MyPy fails completely.  I've tried things like putting them both on the same line but it looks like flake8 and MyPy just don't play nicely together in this sort of conflict.

Worth noting maybe that pylint allows us to disable a particular linting error for an entire function, where AFAICT flake8 does not.

Reply:
that's unfortunate :(

Reply:
> Unfortunately not. We need the type:ignore or else MyPy fails completely. I've tried things like putting them both on the same line but it looks like flake8 and MyPy just don't play nicely together in this sort of conflict.

Just to clarify: https://mypy.readthedocs.io/en/stable/common_issues.html#silencing-linters - this does not work? Is it what you have tried?